HW2

```diff
@@ -0,0 +1,30 @@
+#1. Historic Batting Average
+
+CREATE OR REPLACE TABLE HIS AS SELECT bc.batter as Batter, SUM(bc.Hit) / SUM(bc.atBat) as Average
+FROM batter_counts bc WHERE bc.atBat <> 0 group by bc.batter;
+
+
+#2 Annual Batting Average
+
+CREATE TABLE ANNUAL as SELECT YEAR(g.local_date), bc.batter, SUM(bc.Hit) / SUM(bc.atBat) 
+AS Average FROM batter_counts bc JOIN game g ON bc.game_id = g.game_id
+WHERE bc.atBat <> 0
+GROUP BY YEAR(g.local_date), bc.batter;
+
+
+#3 Rolling
+
+CREATE OR REPLACE TABLE batter_TEST 
+SELECT g.local_date, bc.batter, bc.Hit, bc.atBat
+FROM game g JOIN batter_counts bc 
+ON g.game_id = bc.game_id;
+
+CREATE OR REPLACE TABLE batter_Result 
+SELECT bt.batter, SUM(bt.Hit) / SUM(bt.atBat) AS Average, bt.local_date 
+FROM batter_TEST bt JOIN batter_TEST bt2 ON bt.batter = bt2.batter 
+AND bt2.local_date 
+BETWEEN DATE_SUB(bt.local_date, INTERVAL 100 DAY) AND bt.local_date 
```

Comment:
I would use DATE(local_date) to get the date format because we are calculating the day-rolling average. I don't think local_date is incorrect for the code, though.

Reply:
Thanks!